perf/src/sigverify.rs

```diff
@@ -496,6 +503,43 @@ impl Deduper {
     }
 }
 
+//inplace shrink a batch of packets
+pub fn shrink_batches(batches: &mut Vec<PacketBatch>) -> usize {
+    let mut valid_batch_ix = 0;
+    let mut valid_packet_ix = 0;
+    let mut last_valid_batch = 0;
+    for j in 0..batches.len() {
+        for jj in 0..batches[j].packets.len() {
```

Comment:
Can `j` and `jj` get nicer names too?

Reply:
boo, these are classic C array index names, of the finest vintage

Reply:
I let you have your gross outer-loop-break logic! :sweat_smile: